src/test/kotlin/com/ampnet/reportservice/service/TemplateDataServiceTest.kt

```diff
@@ -328,6 +310,120 @@ class TemplateDataServiceTest : JpaServiceTestBase() {
+                        assertThat(it.revenueShare).isEqualTo(testContext.sharePayout.toEurAmount())
+                        assertThat(it.withdrawals).isEqualTo(testContext.withdraw.toEurAmount())
+                        assertThat(it.balance).isEqualTo(
+                            (
+                                testContext.deposit + testContext.invest -
+                                    testContext.cancelInvestment + testContext.sharePayout - testContext.withdraw
+                                ).toEurAmount()
+                        )
+                    }
+                    secondUserUuid -> {
+                        assertThat(it.deposits).isEqualTo(testContext.deposit.toEurAmount())
+                        assertThat(it.balance).isEqualTo(testContext.deposit.toEurAmount())
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    fun mustThrowExceptionIfUserIsNotAdminOrPlatformManager() {
+        suppose("There are users in the coop") {
+            val firstUser = createUserExtendedResponse(userUuid)
+            val secondUser = createUserExtendedResponse(UUID.randomUUID())
+            val coopResponse = createCoopResponse()
+            val users = createUsersExtendedResponse(listOf(firstUser, secondUser), coopResponse)
+            Mockito.`when`(userService.getAllActiveUsers(userUuid, coop)).thenReturn(users)
+        }
+
+        verify("Service must throw exception if user is not admin or platform manager") {
+            val periodRequest = PeriodServiceRequest(null, null)
+            val user = createUserPrincipal()
+            val exception = assertThrows<InvalidRequestException> {
+                templateDataService.getAllActiveUsersSummaryData(user, periodRequest)
+            }
+            assertThat(exception.errorCode).isEqualTo(ErrorCode.USER_MISSING_PRIVILEGE)
+        }
+    }
+
+    private fun createDepositTx(): TransactionInfo =
+        createTransaction(
+            mintHash, userWalletHash, testContext.deposit.toString(),
+            TransactionType.DEPOSIT
+        )
+
+    private fun createInvestTx(): TransactionInfo =
+        createTransaction(
+            userWalletHash, projectWalletHash, testContext.invest.toString(),
+            TransactionType.INVEST
+        )
+
+    private fun createSharePayoutTx(): TransactionInfo =
+        createTransaction(
+            projectWalletHash, userWalletHash, testContext.sharePayout.toString(),
+            TransactionType.SHARE_PAYOUT
+        )
+
+    private fun createCancelInvestmentTx(): TransactionInfo =
+        createTransaction(
+            projectWalletHash, userWalletHash, testContext.cancelInvestment.toString(),
+            TransactionType.CANCEL_INVESTMENT
+        )
+
+    private fun createWithdrawTx(): TransactionInfo =
+        createTransaction(
+            userWalletHash, burnHash, testContext.withdraw.toString(),
+            TransactionType.WITHDRAW
+        )
```

Comment:
There is no need for all these functions if they are used only once. This can be complete in `createUserTransactionFlow` with the same `createTransaction`.

Reply:
`createDepositTx()` is used twice. I'll remove the other functions